pycbc/tmpltbank/sky_grid.py

```diff
@@ -91,13 +91,19 @@ def read_from_file(cls, path):
             ref_gps_time = hf.attrs['ref_gps_time']
         return cls(ra, dec, detectors, ref_gps_time)
 
-    def write_to_file(self, path):
+    def write_to_file(self, path, extra_attrs, extra_datasets):
         """Writes a sky grid to an HDF5 file."""
         with h5py.File(path, 'w') as hf:
             hf['ra'] = self.ras
             hf['dec'] = self.decs
             hf.attrs['detectors'] = self.detectors
             hf.attrs['ref_gps_time'] = self.ref_gps_time
+            if extra_attrs is not None:
+                for attribute in extra_attrs:
+                    hf.attrs[attribute] = extra_attrs[attribute]
+            if extra_datasets is not None:
+                for datasets in extra_datasets:
+                    hf[datasets] = extra_datasets[datasets]
```

Comment:
This version is a bit more concise (though maybe slightly less readable):
```suggestion
            for attribute in (extra_attrs or {}):
                hf.attrs[attribute] = extra_attrs[attribute]
            for dataset in (extra_datasets or {}):
                hf[dataset] = extra_datasets[dataset]
```